### PR TITLE
[WIP] ANCM + Arm64 + helix

### PR DESCRIPTION
**Pull request by HaoK**
Try to get iis tests running on arm64 helix queue

Tests are running now but most no-op as IIS/IISExpress are not installed

### CODE REVIEW

File: .azure/pipelines/helix-matrix.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # We only want to run full helix matrix on main
-pr: none
+pr:
```
Comment (HaoK) [inline .azure/pipelines/helix-matrix.yml:2]: revert

File: eng/targets/Helix.Common.props [hunk 2]
```diff
@@ -37,27 +37,8 @@
       <!-- aspnetcore-helix-matrix[-internal] (helix-matrix.yml) and RunHelix.ps1 -->
       <!-- aspnetcore-quarantined-tests (quarantined-tests.yml) and RunHelix.ps1 -RunQuarantinedTests -->
       <ItemGroup>
-        <!-- Linux -->
```
Comment (HaoK) [inline eng/targets/Helix.Common.props:40]: revert

**Comment by HaoK** [comment]
@dotnet/aspnet-build how does this tweak look for our arm64 testing, I've asked core-eng to turn on IIS for arm64 so we'll be starting to run tests (once we get things working), but basically I split out the windows arm64 queue into its own group for the arm64 native build and split it out from the other queues

File: .azure/pipelines/helix-matrix.yml [hunk 3]
```diff
@@ -48,3 +52,41 @@ jobs:
     - name: Helix_logs
       path: artifacts/log/
       publishOnError: true
+      
+# Helix ARM64
+- ${{ if ne(variables['System.TeamProject'], 'internal') }}:
```
Comment (dougbu) [inline .azure/pipelines/helix-matrix.yml:57]: Why do this only for public runs e.g. are you dealing w/ another missing internal Helix queue❔ aspnetcore-helix-matrix-internal is disabled in any case.

File: .azure/pipelines/helix-matrix.yml [hunk 4]
```diff
@@ -48,3 +52,41 @@ jobs:
...
+      jobDisplayName: "Tests: Helix ARM64 matrix"
+      agentOs: Windows
+      timeoutInMinutes: 480
+      steps:
+      - script: ./eng/build.cmd
+                -ci
+                -arch x64
+                -pack
+                -all
+        displayName: Build x64/arm64
```
Comment (dougbu) [inline .azure/pipelines/helix-matrix.yml:70]: Need `/p:CrossgenOutput=false` here too; might need it in the next build step as well

File: .azure/pipelines/helix-matrix.yml [hunk 5]
```diff
@@ -48,3 +52,41 @@ jobs:
...
+                -arch arm64
+                -pack
+                -noBuildJava
+                -noBuildNative
+                /p:OnlyPackPlatformSpecificPackages=true
+        displayName: Build ARM64 shared fx
+      - script: .\eng\build.cmd -ci -arch arm64 -noBuildRepoTasks -NoRestore -test -all
+                -projects eng\helix\helix.proj /p:IsHelixWindowsArmCheck=true
+                /p:IsHelixJob=true  /p:RunTemplateTests=true
+                /p:CrossgenOutput=false /p:ASPNETCORE_TEST_LOG_DIR=artifacts/log
```
Comment (dougbu) [inline .azure/pipelines/helix-matrix.yml:83]: Generally, be consistent w/ property settings in all three build steps. This line appears multiple times in the other job for example.

**Comment by japui** [comment]
WIP] ANCM + Arm64 + hélice #

**Pull request closed without merging**